Make root project an aggregation of the other projects

paulmr opened the pull request:
This means that you can now publish all of the deliverables from the different projects by doing a `+publishSigned` (or a `+release` etc) from the root project, instead of publishing each individual project individually.

steppenwells commented:
I thought the release plug handled cross versions for you. Also it might be useful to add a note to the readme about the release process.

Other than that I can't see anything wrong with the changes, I'm happy with them if they work as expected.

theefer commented:
I noticed it seems to crossbuild automatically for panda, possibly thanks to [explicit configuration](https://github.com/guardian/pan-domain-authentication/blob/master/project/PanDomainAuthenticationBuild.scala#L48) (see also a few lines below)?

paulmr commented:
sbt release seems to be broken with aggregated projects because `runAggregated` for some reason doesn't run the actual taks:

https://github.com/sbt/sbt/issues/1210

cb372 commented:
If you're trying to do cross-building for some subprojects and not for others, this friendly shiba can help you https://github.com/sbt/sbt-doge

cb372 commented:
Also, I think @steppenwells and @theefer are right: if you add `ReleaseKeys.crossBuild := true`, you should be able to just run `sbt release`. `sbt +release` is fraught with problems, e.g. updating the version once, releasing for 2.11, then bumping the version again before releasing for 2.10.

Also in the release steps you should use `publishArtifacts` instead of `releaseStepTask(PgpKeys.publishSigned)`. To tell it to sign the artifacts, you need to add `releasePublishArtifactsAction := publishSigned.value` somewhere.

paulmr commented:
Thanks @cb372 . Is there a solution for the `runAggregated` bug though?

paulmr commented:
Or is `publishArtifacts` a `ReleaseStep` instead of a task?

theefer commented:
@cb372 Quite disappointing to discover that @bmjames is not the author of sbt-doge.

paulmr commented:
That does the trick, thanks @cb372 !

paulmr commented:
I have updated the build to use sbt-release's cross building capabilities, and added some documentation on how to publish.

How does it look now @steppenwells and @cb372 ?

Thanks.

mchv commented:
The changes looks good but why do we have several projects at the first place? we could just include the thrift definition file exactly like in [story-package model](https://github.com/guardian/story-packages-model/blob/master/build.sbt)?

mchv commented:
I am going to merge this an plan to release a new version tomorrow.

mchv commented:
> The changes looks good but why do we have several projects at the first place? we could just include the thrift definition file exactly like in story-package model?

I have created https://github.com/guardian/content-atom/pull/17 which include the thrift file in the default artefact (you can still use the `src` if you just want the source)
